src/deal-tracking/DealTrackingService.ts

```diff
@@ -31,19 +30,116 @@ export default class DealTrackingService extends BaseService {
+        await this.insertDealFromFilscan(client, lastDeal.length > 0 ? lastDeal[0].dealId! : 0);
       } catch (error) {
-        this.logger.error('Encountered an error when importing deals from filfox', { error });
+        this.logger.error('Encountered an error when importing deals from filescan', error);
       }
       try {
         await this.updateDealFromLotus(client);
       } catch (error) {
-        this.logger.error('Encountered an error when updating deals from lotus', { error });
+        this.logger.error('Encountered an error when updating deals from lotus', error);
       }
     }
   }
 
-  private async insertDealFromFilfox (client: string, lastDeal: number): Promise<void> {
+  /**
+   * Read from filscan api for PublishStorageDeal status
+   *
+   * @param client
+   * @param lastDeal
+   */
+  private async insertDealFromFilscan (client: string, lastDeal: number): Promise<void> {
+    this.logger.debug('updating deals from filscan', { client, lastDeal });
+    let page = 0;
+    let response;
+    do {
+      const offset = page * 25;
+      let breakOuter = false;
+
+      let url = 'https://api.filscan.io:8700/rpc/v1';
+      if (client.startsWith('t')) {
+        url = 'https://calibration.filscan.io:8700/rpc/v1';
+      }
+      this.logger.debug(`Fetching from ${url}`);
+      response = await axios.post(url, {
+        id: 1,
+        jsonrpc: '2.0',
+        params: [client, offset, 25],
+        method: 'filscan.GetMarketDeal'
+      }, {
+        headers: {
+          'content-type': 'application/json'
+        }
+      });
+      if (Array.isArray(response.data['result']['deals'])) {
+        const jsonResult = response.data['result'];
+        this.logger.debug(`Received ${offset} out of ${jsonResult['total']} deal entries.`);
+        for (const deal of jsonResult['deals']) {
+          if (deal['dealid'] <= lastDeal || jsonResult['deals'].length < 25) {
+            breakOuter = true;
+            this.logger.info('Reached to the last page or last checked deal.');
+            break;
+          }
+          const publishedDeal = await Datastore.DealStateModel.findOne({
+            dealId: deal['dealid']
+          });
+          if (publishedDeal) {
+            continue;
+          }
+          const proposedDeal = await Datastore.DealStateModel.findOne({
+            pieceCid: deal['piece_cid'],
+            provider: deal['provider'],
+            client: deal['client'],
+            state: 'proposed'
+          });
+
+          if (proposedDeal) {
+            await Datastore.DealStateModel.updateOne({
+              _id: proposedDeal._id
+            }, {
+              $set: {
+                dealId: deal['dealid'],
+                state: 'published'
+              }
+            });
+            this.logger.debug(`Deal ${deal['dealid']} was proposed through singularity. Filling in deal ID.`);
+          } else {
+            await Datastore.DealStateModel.updateOne({
```

Comment:
I think you can use create here since you won't find the entries with that dealId according to logic on line 87